rules.mk

```diff
@@ -57,18 +60,25 @@ $(generated_files): bin/goimports
 
 # this is quick and its really only for rebuilding when dev'ing, I wish go would
 # output deps in make syntax like gcc does... oh well this is good enough
-cmd/boots/boots: $(shell git ls-files | grep -v -e vendor -e '_test.go' | grep '.go$$' ) ipxe/bindata.go syslog/facility_string.go syslog/severity_string.go
+cmd/boots/boots: $(shell git ls-files | grep -v -e vendor -e '_test.go' | grep '.go$$' ) build_all_ipxe go_generate syslog/facility_string.go syslog/severity_string.go
 	go build -v -ldflags="-X main.GitRev=${GitRev}" -o $@ ./cmd/boots/
 
 include ipxev.mk
 ipxeconfigs := $(wildcard ipxe/ipxe/*.h)
 
-ipxe/bin/ipxe.efi: ipxe/ipxe/build/bin-x86_64-efi/ipxe.efi
-ipxe/bin/snp-nolacp.efi: ipxe/ipxe/build/bin-arm64-efi/snp.efi
-ipxe/bin/undionly.kpxe: ipxe/ipxe/build/bin/undionly.kpxe
-ipxe/bin/ipxe.efi ipxe/bin/snp-nolacp.efi ipxe/bin/undionly.kpxe:
+# copy ipxe binaries into location available for go embed
+tftp/ipxe/ipxe.efi: ipxe/ipxe/build/bin-x86_64-efi/ipxe.efi
+tftp/ipxe/snp-nolacp.efi: ipxe/ipxe/build/bin-arm64-efi/snp.efi
+tftp/ipxe/undionly.kpxe: ipxe/ipxe/build/bin/undionly.kpxe
+tftp/ipxe/ipxe.efi tftp/ipxe/snp-nolacp.efi tftp/ipxe/undionly.kpxe:
+	mkdir -p tftp/ipxe
 	cp $^ $@
 
+tftp/ipxe/snp-hua.efi:
```

Comment:
Can we use a non-code related path here maybe? Like `ipxe-builds` or `built-assets` or ... anything else really? Or does it need to be within `tftp` because `go:embed ..` doesn't work?

Reply:
yeah, from all my testing it needs to be within the `tftp` directory for `go:embed` to work.

Reply:
gotcha